### DISCUSSION TITLE
[2일차 - 예진] var, let, const 의 차이점

### DISCUSSION THREAD

**Discussion by ez1n** [zzangCS/zzangInterview#10] [프론트 면접 질문]
var, let, const 의 차이점을 설명해 주세요.

**Comment by smarfy99** [comment]
var 는 변수 중복 선언이 가능하고, 함수 외부에서 선언한 변수는 모두 전역 변수가 되며, 변수 선언 이전에 변수를 참조하면 undefined를 반환한다. 
스코프를 최대한 좁혀 사이드이펙트를 최소화하기 위해 ES6 부터 나온 let은 변수 중복 선언은 불가하지만, 재할당은 가능하다. const는 선언과 초기화가 동시에 진행되어야 하며, 재선언, 재할당이 불가하다.

**Reply by smarfy99** [reply]
https://www.howdy-mj.me/javascript/var-let-const

**Comment by skyhanull** [comment]
var는 재선언,재할당이 모두 가능하다

let은 재선언은 불가능하지만 재 할당이 가능하다

const는 재선언,재할당 모두 불가능 하다

**Comment by ez1n** [comment]
**var** 
1. 선언 단계와 초기화 단계가 동시에 진행되며 값을 할당하지 않은 경우 undefined를 할당하여 초기화한다.
2. 중복 선언이 가능하고 새로운 값을 재할당 할 수 있다.

**let** 
1. 중복 선언이 불가하다.
2. 재할당이 가능하다.

**const**
1. 선언과 초기화를 동시에 진행해야 한다.
2. 재선언 및 재할당이 불가하다.
⇒ 객체의 경우 재할당이 가능함
